mapie/calibration.py

```diff
@@ -0,0 +1,559 @@
+           highest score, see section 2 of [1].
+
+        By default "top_label".
+
+    calibrator : Optional[Union[str, RegressorMixin]]
+        Any calibrator with scikit-learn API
+        (i.e. with fit, predict, and predict_proba methods), by default
+        ``None``.
+        If ``None``, calibrator defaults to a string ``sigmoid`` instance.
+
+        By default ``None``.
+
+    cv: Optional[str]
+        The cross-validation strategy for computing scores :
+
+        - ``split``, performs a standard splitting procedure into a
+          calibration and test set.
+        - ``"prefit"``, assumes that ``estimator`` has been fitted already.
+          All data provided in the ``fit`` method is then used
+          to calibrate the predictions through the score computation.
+          At prediction time, quantiles of these scores are used to estimate
+          prediction sets.
+
+        By default ``split``.
+
+    Attributes
+    ----------
+    valid_methods: List[str]
+        List of all valid methods.
+
+    classes_: NDArray
+        Array with the name of each class.
+
+    n_classes_: int
+        Number of classes that are in the training dataset.
+
+    uncalib_pred: NDArray
+        Array of the uncalibrated predictions.
+
+    calibrators: Dict[str, RegressorMixin]
+        Dictionnary of all the fitted calibrators.
+
+    References
+    ----------
+    [1] Gupta, Chirag, and Aaditya K. Ramdas. "Top-label calibration
+    and multiclass-to-binary reductions." arXiv preprint
+    arXiv:2107.08353 (2021).
+
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> from mapie.calibration import MapieCalibrator
+    >>> X_toy = np.arange(9).reshape(-1, 1)
+    >>> y_toy = np.stack([0, 0, 1, 0, 1, 2, 1, 2, 2])
+    >>> mapie = MapieCalibrator().fit(X_toy, y_toy, random_state=20)
+    >>> y_calib = mapie.predict_proba(X_toy)
+    >>> print(y_calib)
+    [[0.84900723 0.         0.        ]
+     [0.75432411 0.         0.        ]
+     [0.62285341 0.         0.        ]
+     [0.         0.33333333 0.        ]
+     [0.         0.33333333 0.        ]
+     [0.         0.33333333 0.        ]
+     [0.         0.         0.33333002]
+     [0.         0.         0.54326683]
+     [0.         0.         0.66666124]]
+    """
+
+    fit_attributes = [
+        "estimator",
+        "calibrators",
+    ]
+
+    valid_calibrators = {
+        "sigmoid": _SigmoidCalibration(),
+        "isotonic": IsotonicRegression(out_of_bounds="clip")
+    }
+
+    valid_methods = ["top_label"]
+
+    def __init__(
+        self,
+        estimator: Optional[ClassifierMixin] = None,
+        method: str = "top_label",
+        calibrator: Optional[Union[str, RegressorMixin]] = None,
+        cv: Optional[str] = "split",
+    ) -> None:
+        self.estimator = estimator
+        self.method = method
+        self.calibrator = calibrator
+        self.cv = cv
+
+    def _fit_calibrator(
```

Comment:
same confusion about calibrator

Reply:
I wouldn't think so, I think there is a calibrator, that's the estimator used to calibrate the outputs of the classifier.